File: backend/databasehandler/src/main/java/com/pixtranslator/backend/databasehandler/model/GermanWord.java
package com.pixtranslator.backend.databasehandler.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
public class GermanWord implements Serializable {

  @Id
  @Getter
  @Setter
  @Column(unique = true)
  private String word;

  @OneToMany(fetch = FetchType.LAZY, mappedBy = "germanword")
  @JsonManagedReference
  @Getter
  @Setter
  private List<EnglishWord> englishWords;

}

File: backend/databasehandler/src/main/kotlin/com/pixtranslator/backend/databasehandler/model/GermanWord.kt
package com.pixtranslator.backend.databasehandler.model

import com.fasterxml.jackson.annotation.JsonManagedReference
import lombok.Getter
import lombok.Setter
import java.io.Serializable
import javax.persistence.*

@Entity
class GermanWord : Serializable {
  @Id
  @Column(unique = true)
  var word: String? = null

  @OneToMany(fetch = FetchType.LAZY, mappedBy = "germanword")
  @JsonManagedReference
  val englishWords: List<EnglishWord> = mutableListOf()
}
